tests/unittests/algo/mofa/test_mofa.py

```diff
@@ -0,0 +1,126 @@
+"""Perform integration tests for `orion.algo.mofa`."""
+from __future__ import annotations
+
+from typing import ClassVar
+
+import pytest
+import scipy
+from packaging import version
+
+from orion.algo.mofa.mofa import MOFA
+from orion.algo.space import Space
+from orion.testing.algo import BaseAlgoTests, TestPhase
+
+
+@pytest.fixture(autouse=True)
+def _config(request):
+    """Fixture to xfail test_cat_data for 2nd and 3rd runs."""
+    if "num" not in request.fixturenames:
+        yield
+        return
+
+    test_name, _ = request.node.name.split("[")
+    if test_name == "test_cat_data" and request.getfixturevalue("num") > 0:
+        pytest.xfail("MOFA does not explore well categorical dimensions")
+
+    yield
+
+
+def test_scipy_version():
+    try:
+        algo = MOFA(Space())
+    except RuntimeError:
+        assert version.parse(scipy.__version__) < version.parse("1.8")
+
+
+@pytest.mark.skipif(
+    version.parse(scipy.__version__) < version.parse("1.8"),
+    reason="requires scipy v1.8 or higher",
+)
+class TestMOFA(BaseAlgoTests):
+    """Test suite for algorithm MOFA"""
+
+    algo_name = "mofa"
+
+    config = {
+        "seed": 1234,  # Because this is so random
+        "index": 1,
+        "n_levels": 5,
+        "strength": 2,
+        "threshold": 0.1,
+    }
+
+    n_trials = config["index"] * config["n_levels"] ** config["strength"]
+
+    phases: ClassVar[list[TestPhase]] = [
+        TestPhase("1st-run", 0, "space.sample"),
+        TestPhase("2nd-run", n_trials, "space.sample"),
+        TestPhase("3rd-run", n_trials * 2, "space.sample"),
+    ]
+
+    def get_num(self, num):
```

Comment:
```suggestion
    @classmethod
    def get_num(cls, num):
```
This should solve it @wagnersj